api/api.go

```diff
@@ -0,0 +1,163 @@
+package api
+
+import (
+	"doccer/model"
+	"encoding/json"
+	"github.com/gorilla/mux"
+	"net/http"
+)
+
+type Api struct {
+	useCases model.UseCasesInterface
+}
+func NewApi(x model.UseCasesInterface) *Api {
+	return &Api{
+		useCases: x,
+	}
+}
+
+func (a *Api) Router() http.Handler {
+	router := mux.newRouter
+
+	router.HandleFunc("/register", a.register).Methods(http.MethodPost)
+	router.HandleFunc("/login", a.login).Methods(http.MethodPost)
+	router.HandleFunc("/logout", a.logout).Methods(http.MethodPost)
+
+	router.HandleFunc("/docs/{doc_id}", a.getDoc).Methods(http.MethodGet)
+	router.HandleFunc("/docs/{doc_id}", a.createDoc).Methods(http.MethodPost)
+	router.HandleFunc("/docs/{doc_id}", a.deleteDoc).Methods(http.MethodDelete)
+	router.HandleFunc("/docs/{doc_id}", a.editDoc).Methods(http.MethodPut)
+
+	router.HandleFunc("/docs/get_all_docs", a.getAllDocs).Methods(http.MethodGet)
+
+	router.HandleFunc("/users", a.editUser).Methods(http.MethodPut)
+	router.HandleFunc("/users", a.getUser).Methods(http.MethodGet)
+
+	router.HandleFunc("/users/friends", a.getFriends).Methods(http.MethodGet)
```

Comment:
Is this method for retrieving all friends for a specific user?

Reply:
No, user's friends aren't visible for other users